### PR TITLE
BigInt valueOf tests

### PR DESCRIPTION
**Pull request by cxielarko**
Tests for the BigInt valueOf method

### CODE REVIEW

**Comment by rwaldron** [comment]
@cxielarko needs rebase

**Comment by cxielarko** [comment]
I rebased this branch and added missing feature flags

File: test/built-ins/BigInt/prototype/valueOf/this-value-err.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+/*---
+esid: sec-bigint.prototype.valueof
+description: BigInt.prototype.valueOf this value coercion
+info: >
+  BigInt.prototype.valueOf ( )
+
+  Return ? thisBigIntValue(this value).
+includes: [typeCoercion.js]
+features: [BigInt, arrow-function, Symbol, Symbol.toPrimitive]
```
Comment (leobalter) [inline test/built-ins/BigInt/prototype/valueOf/this-value-err.js]: this seems to have more tagged features than necessary. Is Symbol.toPrimitive ever used?

File: test/built-ins/BigInt/prototype/valueOf/this-value.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  Return ? thisBigIntValue(this value). 
+includes: [typeCoercion.js]
+features: [BigInt, arrow-function, Symbol, Symbol.toPrimitive]
+---*/
+
+testCoercibleToBigIntThisValue(
+  0n,
+  (x) => assert.sameValue(0n, BigInt.prototype.valueOf.call(x))
+);
```
Comment (leobalter) [inline test/built-ins/BigInt/prototype/valueOf/this-value.js]: static direct calls would provide a better reading for this. Expanding the text for readability can help here.

**Pull request merged**